service/testing/runner_test.cc

```diff
@@ -616,3 +619,22 @@ BOOST_AUTO_TEST_CASE( test_set_prctl_from_thread )
     BOOST_CHECK_EQUAL(runResult.returnCode, 0);
     BOOST_CHECK_EQUAL(runResult.signum, -1);
 }
+#endif
+
+#if 1
+/* This test ensures that onTerminate is called with the appropriate RunResult
+ * when the runWrapper process fails. */
+BOOST_AUTO_TEST_CASE( test_unexisting_runner_helper )
```

Comment:
Can you confirm that prior to the fix the deadlock was reproduced with that test?